app/src/main/java/com/github/kpdandroid/utils/camera/SnapshotAnalyzer.kt

```diff
@@ -0,0 +1,54 @@
+package com.github.kpdandroid.utils.camera
+
+import android.graphics.Bitmap
+import android.util.Log
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageProxy
+import androidx.compose.ui.geometry.Offset
+import com.github.kpdandroid.utils.detection.detectTimed
+import com.github.kpdandroid.utils.rgbaBytesToBitmap
+import com.github.kpdandroid.utils.rgbaBytesToRgbBytes
+import com.github.kpdlib.KeypointDetector
+import java.nio.ByteBuffer
+
+private const val TAG = "SnapshotAnalyzer"
+
+class SnapshotAnalyzer(
+    private val detectorGetter: () -> KeypointDetector?,
+    private val onResult: (snapshot: Bitmap, keypoints: List<Offset>, calcTimeMs: Long) -> Unit
+) : ImageAnalysis.Analyzer {
+    override fun analyze(image: ImageProxy) {
+        val width = image.width
+        val height = image.height
+        val (snapshot, rowStride, pixelStride) =
+            image.planes[0].run { Triple(buffer.toByteArray(), rowStride, pixelStride) }
+        Log.v(
+            TAG,
+            "Analyzing snapshot:\n" +
+                "- size: ${width}x$height\n" +
```

Comment:
Again, this is because without them we'll get `widthx`. I doubt that adding braces to other variables will improve readability: seems like it'll make the string more cumbersome to read.